.circleci/config.yml

```diff
@@ -0,0 +1,50 @@
+version: 2.1
+
+jobs:
+  build:
+    docker:
+      - image: cimg/node:15.11.0
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - dependency-cache-1-{{ checksum "package-lock.json" }}
+
+      - run: npm install
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: dependency-cache-1-{{ checksum "package-lock.json" }}
+
+      - persist_to_workspace:
+          root: "."
+          paths:
+            - node_modules
+
+  test:
+    docker:
+      - image: cimg/node:15.11.0
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+
+      - attach_workspace:
+          at: "."
+
+      - run: npm run test
+
+workflows:
+  version: 2
+  commit:
+    jobs:
+      - build
+      - test:
+          requires:
+            - build
```

Comment:
Yeah it shouldn't be _necessary_; I imagine it will just save the builds some time to get students quicker feedback. If you're on a paid plan it will save some build minutes -> $.

Reply:
We are bumping up against the limits of the free plan so this won't hurt